### PR TITLE
only compile clvm for developers

### PR DESCRIPTION
**Pull request by altendky**
Set the `CHIA_DEV_COMPILE_CLVM_ON_IMPORT` environment variable to anything except `""` to trigger automatic recompilation of the Chialisp on load.  This will also be triggered when running tests based on the pytest module having been loaded.

Should help with https://github.com/Chia-Network/chia-blockchain/issues/13209.

Draft for:
- [ ] General Chialisp developer input

### CODE REVIEW

File: chia/wallet/puzzles/load_clvm.py [hunk 1]
```diff
@@ -86,36 +89,59 @@ def load_serialized_clvm(
...
     clvm_blob = bytes.fromhex(clvm_hex)
     return SerializedProgram.from_bytes(clvm_blob)
 
 
-def load_clvm(clvm_filename, package_or_requirement=__name__, include_standard_libraries: bool = False) -> Program:
+def load_clvm(
+    clvm_filename,
+    package_or_requirement=__name__,
+    include_standard_libraries: bool = False,
+    recompile: bool = True,
```
Comment (arvidn) [inline chia/wallet/puzzles/load_clvm.py:122]: is there a good reason to not just default this to?
```
recompile: bool = recompile_requested,
```

And not bother to introduce a new name `load_clvm_maybe_recompile()`. Are there still use cases for the function where we try to recompile unconditionally?

If so, I would still think that would be the uncommon case that should have a longer name.

File: tools/manage_clvm.py [hunk 2]
```diff
@@ -164,4 +181,52 @@ def main() -> int:
     return 1 if overall_fail else 0
 
 
+@main.command()
+def build() -> int:
```
Comment (arvidn) [inline tools/manage_clvm.py:185]: is this function triggered by CI? i.e. is it demonstrated to work and produce identical output?

**Review by arvidn — COMMENTED** [review]
looks reasonable. It seems reasonable to me to leave the `load_clvm()` name as-is, and change its behavior to not, unconditionally, attempt a recompilation. It would make the patch smaller and lower risk or conflicts with other PRs.

At a higher level, it seems odd to me to have this logic in `load_clvm()` to begin with. Building and hashing of the clvm programs would make a lot more sense to do in pre-commit, never at import-time. Is that general consensus and this is a step in that direction, or is there a good reason to do it at import-time?

File: tools/manage_clvm.py (hunk 2 above)
Comment (altendky) [reply tools/manage_clvm.py:185]: Nope.  This was a side step that wasn't removed.

File: chia/wallet/puzzles/load_clvm.py (hunk 1 above)
Comment (altendky) [reply chia/wallet/puzzles/load_clvm.py:122]: The concern was that other projects might be using it.  The fact it is buried down in our wallet puzzles directory seems both bad and an indication against that.  @trepca brought up the concern.

**Comment by altendky** [comment]
I agree that this shouldn't be an import-time activity at all.  I have been peeling off pieces around this topic for awhile to clean it up and yes, this is just the next piece.  The goal here is to at least relegate the compilation to "developers".  Though a compromise was made to trigger it when running tests as well so that chialisp devs don't have to set an environment variable nor otherwise trigger compilation in their workflows.

File: chia/wallet/puzzles/load_clvm.py (hunk 1 above)
Comment (arvidn) [reply chia/wallet/puzzles/load_clvm.py:122]: although, come to think of it, the NFT wallet does de-compile and re-compile clvm code. I'm not sure why, but there is a dependency there too.

**Review by arvidn — APPROVED** [review]

**Review by trepca — APPROVED** [review]

File: chia/wallet/puzzles/load_clvm.py (hunk 1 above)
Comment (trepca) [reply chia/wallet/puzzles/load_clvm.py:122]: It'd be great to separate into dev libraries, but currently even `chia-dev-tools` depends on `chia-blockchain`. For any serious coin development, developers need access to simulator which is in `chia-blockchain` too and all the standard puzzles.

**Pull request merged**